### PR TITLE
Incorporate Elan's feedback

### LINKED ISSUE #1: [main issue] Improve the optimisation practical
# Sub issues

- [x] Fix errors #2 
* Fix pseudocode (errors in Adam section)

- [x] Clarity of descriptions / text #3
* Update language used to be more friendly to non-native English speakers
* Ensure that main practical goals are clearly met
* Tweak some descriptions for better clarity
* Ensure all relevant terminology is introduced prior to it being needed (or at least link to a source)

- [x] Clarity / comparability of methods #4 
* Consistency to starting x and y of Banana function section
* Update SGD with decay hyper-parameters so that this method is somewhat competitive in the “Putting it all into practice” section
* Fix code variable naming/notation to be consistent (learning rate decay is different to all other sections)
* Pseudocode notation to be consistent (spacing around “*”, variable names, more pseudocode like - less python-like, etc)

- [x] Increase intuition #5 
* Add more interactive elements
	* “what are gradients” section
	* practically to see the effect of batch size

- [ ] Answers and solutions at the ready #6 

- [x] Aesthetics #7 
* Enlarge the global minimum star in the plot

- [ ] Add comments to the main code #28

### PR DESCRIPTION
**Pull request by ElanVB**
# If you are happy with the state of the `improve optimisation` branch (the changes in this pull-request), please merge! :)
## The issues that this pull-request addresses can be found here: ElanVB/indaba-pracs-2019#1

Resolves ElanVB/indaba-pracs-2019#1

### CODE REVIEW

**Comment by ElanVB** [comment]
@JamesAllingham Feel free to merge if you are happy with it :)

**Comment by JamesAllingham** [comment]
Thanks for the hard work Elan, this looks great and definitely improves the practical in a big way!
